src/Transport/Sending/MessageDispatcher.cs

```diff
@@ -65,18 +67,20 @@ public Task Dispatch(TransportOperations outgoingMessages, TransportTransaction
             foreach (var transportOperation in multicastTransportOperations)
             {
                 var receiverConnectionAndPathToUse = transportOperation.RequiredDispatchConsistency == DispatchConsistency.Isolated ? (null, null) : receiverConnectionAndPath;
+                var transactionToUse = transportOperation.RequiredDispatchConsistency == DispatchConsistency.Isolated ? null : committableTransaction;
 
                 var sender = messageSenderPool.GetMessageSender(topicName, receiverConnectionAndPathToUse);
 
                 try
                 {
                     var message = transportOperation.Message.ToAzureServiceBusMessage(transportOperation.DeliveryConstraints, partitionKey);
 
-                    using (var scope = CreateTransactionScope(transportOperation.RequiredDispatchConsistency, shouldSuppressTransaction))
+                    using (var scope = transactionToUse.ToScope())
                     {
                         // Invoke sender and immediately return it back to the pool w/o awaiting for completion
                         tasks.Add(sender.SendAsync(message));
-                        scope?.Complete();
+
+                        scope.Complete();
```

Comment:
Would you fancy a comment here that explains the committable tx will not be committed because this scope is not the owner?